src/librustc/traits/coherence.rs

```diff
@@ -148,6 +154,11 @@ pub fn trait_ref_is_knowable<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
     orphan_check_trait_ref(tcx, trait_ref, InferIsLocal(true)).is_err()
 }
 
+pub fn trait_ref_is_local_or_fundamental<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
+                                                         trait_ref: &ty::TraitRef<'tcx>) {
```

Comment:
Rollup failed due to this PR. This function should return `bool`.

```rust
[00:06:18] error[E0600]: cannot apply unary operator `!` to type `()`
[00:06:18]    --> /checkout/src/librustc/traits/coherence.rs:143:8
[00:06:18]     |
[00:06:18] 143 |     if !trait_ref_is_local_or_fundamental(tcx, trait_ref) {
[00:06:18]     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:06:18] 
[00:06:18] error[E0308]: mismatched types
[00:06:18]    --> /checkout/src/librustc/traits/coherence.rs:159:5
[00:06:18]     |
[00:06:18] 158 |                                                          trait_ref: &ty::TraitRef<'tcx>) {
[00:06:18]     |                                                                                          - help: possibly return type missing here?: `-> bool `
[00:06:18] 159 |     trait_ref.def_id.krate == LOCAL_CRATE || tcx.has_attr(trait_ref.def_id, "fundamental")
[00:06:18]     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected (), found bool
[00:06:18]     |
[00:06:18]     = note: expected type `()`
[00:06:18]                found type `bool`
[00:06:18] 
[00:06:18] error[E0308]: mismatched types
[00:06:18]     --> /checkout/src/librustc/traits/select.rs:1097:78
[00:06:18]      |
[00:06:18] 1097 |                                                                              trait_ref) {
[00:06:18]      |                                                                              ^^^^^^^^^ expected reference, found struct `ty::sty::TraitRef`
[00:06:18]      |
[00:06:18]      = note: expected type `&ty::sty::TraitRef<'_>`
[00:06:18]                 found type `ty::sty::TraitRef<'_>`
[00:06:18]      = help: try with `&trait_ref`
[00:06:18] 
[00:06:18] error[E0600]: cannot apply unary operator `!` to type `()`
[00:06:18]     --> /checkout/src/librustc/traits/select.rs:1096:32
[00:06:18]      |
[00:06:18] 1096 |                   let cause = if !coherence::trait_ref_is_local_or_fundamental(self.tcx(),
[00:06:18]      |  ________________________________^
[00:06:18] 1097 | |                                                                              trait_ref) {
[00:06:18]      | |_______________________________________________________________________________________^
[00:06:18] 
[00:06:22] error: aborting due to 4 previous errors
[00:06:22] 
[00:06:22] error: Could not compile `rustc`.
```

Reply:
Guess I forgot to run the tests! Good ol' bors.